packages/grid/src/Grid.tsx

```diff
@@ -1206,10 +1209,18 @@ class Grid extends PureComponent<GridProps, GridState> {
         newCursorRow = null;
       }
 
+      const selectionChanged =
+        newSelectedRanges.length !== selectedRanges.length ||
+        newSelectedRanges.some(
+          (range, index) => !range.equals(selectedRanges[index])
+        );
+
       return {
         cursorRow: newCursorRow,
         cursorColumn: newCursorColumn,
-        selectedRanges: newSelectedRanges,
+        // If the selection ranges are non-overlapping, then selectedRanges already includes everything
```

Comment:
What does "non-overlapping" mean here? Is that synonymous with all ranges are unchanged?